test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -2564,6 +2470,26 @@ func initTestSchedulerForFrameworkTest(t *testing.T, testCtx *testutils.TestCont
 	return testCtx
 }
 
+// cleanupTestSchedulerForFrameworkTest will close the scheduler and delete the nodes and pods.
+// If you want to tear down the whole context, call `testutils.CleanupTest(t, testCtx)` instead.
+func cleanupSchedulerFrameworkTest(t *testing.T, testCtx *testutils.TestContext) {
```

Comment:
This mentions nodes and pods. Why just those? What about deleting namespaces?

The comment explains the "what" (which can be seen from the code), but not the less obvious "why". Can you add which objects need to be deleted here, now and in the future?

Reply:
Instead of having some code elsewhere create something and adding separate cleanup code for it here, can we make it so that the code which creates something also has the corresponding cleanup code and registers a callback for it?

This would be conceptually the same as `gingko.DeferCleanup`.

Reply:
Update the comment and make the teardown func an output parameter. As you expected?

Reply:
This is the shutdown function returned by `initTestSchedulerForFrameworkTest`, right? That looks like a better approach than the stand-alone `cleanupTestSchedulerForFrameworkTest` function here.